Kitodo/src/test/java/org/kitodo/production/services/data/RulesetServiceIT.java

```diff
@@ -106,6 +107,16 @@ public void shouldFindByTitle() throws DAOException {
         assertEquals(1, rulesetService.getByTitle(slubDD).size(), rulesetNotFound);
     }
 
+    @Test
+    public void shouldFindByTitleWithClientEagerlyLoaded() throws DAOException {
+        List<Ruleset> rulesets = rulesetService.getByTitleWithClient(slubDD);
+        assertEquals(1, rulesets.size(), rulesetNotFound);
+
+        Ruleset ruleset = rulesets.get(0);
+        assertNotNull(ruleset.getClient(), "Client should be eagerly loaded");
+        assertNotNull(ruleset.getClient().getId(), "Client ID should be accessible without exception");
```

Comment:
If you want to test if no exception is thrown than use something like that

```java
assertDoesNotThrow(() -> {
        ruleset.getClient().getId();
    });
```
See https://www.baeldung.com/junit-assert-exception

Adding `throws DAOException` to method declaration let the test not fail if this exception is thrown.